File: javascript/easy/series.js
export class Series {
  constructor(series) {
    if (!series) {
      throw new Error("series cannot be empty");
    }
    this.series = series;
  }

  slices(sliceLength) {
    if (sliceLength > this.series.length) {
      throw new Error("slice length cannot be greater than series length");
    } else if (sliceLength === 0) {
      throw new Error("slice length cannot be zero");
    } else if (sliceLength < 0) {
      throw new Error("slice length cannot be negative");
    }
    let collection = [];
    for (let i = 0; i <= this.series.length - sliceLength; i++) {
      let newSlice = [];
      for (let j = 0; j < sliceLength; j++) {
        newSlice.push(+this.series[i + j]);
      }
      collection.push(newSlice);
    }
    return collection;
  }
}

File: python/easy/series.py
def slices(series, length):
    if series == "":
        raise ValueError("series cannot be empty")
    elif length == 0:
        raise ValueError("slice length cannot be zero")
    elif length < 0:
        raise ValueError("slice length cannot be negative")
    elif length > len(series):
        raise ValueError("slice length cannot be greater than series length")
    return [series[i:i + length] for i in range(len(series) + 1 - length)]
